File: c++/leastInterval.cpp
class Solution {
public:
    int leastInterval(vector<char>& tasks, int n) {
        int mx = 0, mxCount = 0;
        vector<int> count(26, 0); 
        for (auto t:tasks){
            count[t - 'A']++;
        }
        sort(count.begin(), count.end());
        mx = count[25];
        for (int i = 0; i < 25; i ++){
            if (count[25-i] == mx){
                mxCount++;
            }
        }
        int taskSlot = (mx - 1) * (n + 1 - mxCount);
        int mxTaskSize = mx * mxCount;
        int taskLeft = tasks.size() - mxTaskSize;
        int extra = std::max(0, taskLeft - taskSlot );
        return mxTaskSize + taskSlot + extra;
    }
};

File: python/LeastInterval.py
class Solution:
    def leastInterval(self, tasks, n):
        """
        :type tasks: List[str]
        :type n: int
        :rtype: int
        """
        total = len(tasks)
        count = collections.Counter(tasks)
        count_sort = count.most_common()
        max_freq = count_sort[0][1]
        max_tasks = 0
        for item in count_sort:
            if item[1] == max_freq:
                max_tasks += 1
            else:
                break

        available_slot = (max_freq - 1) * (n + 1 - max_tasks)
        remain = total - max_freq * max_tasks
        if remain > available_slot:
            return max_freq * max_tasks + remain
        else:
            return max_freq * max_tasks + available_slot
